### PR TITLE
Add search with tests for sciname on ncbi taxa

### PR DESCRIPTION
**Pull request by jayrbolton**

### CODE REVIEW

File: stored_queries/ncbi_tax/ncbi_taxon_search_sci_name.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Search for an ncbi taxon with a scientific name
+# Results are limited to 10k
```
Comment (MrCreosote) [inline stored_queries/ncbi_tax/ncbi_taxon_search_sci_name.yaml]: Doesn't looks like they are limited to 10k? There probably should be limits and offset and limit though, things can get real bad based on personal experience if either of those are too large

File: test/stored_queries/test_ncbi_tax.py [hunk 2]
```diff
@@ -136,3 +136,52 @@ def test_siblings_nonexistent_node(self):
             data=json.dumps({'key': 'xyz'}),  # Nonexistent node
         ).json()
         self.assertEqual(resp['count'], 0)
+
+    def test_search_sciname_prefix(self):
+        """Test a query to search sciname."""
+        resp = requests.post(
+            _CONF['re_api_url'] + '/api/v1/query_results',
+            params={'stored_query': 'ncbi_taxon_search_sci_name'},
+            data=json.dumps({'search_text': 'prefix:bact'}),
```
Comment (MrCreosote) [inline test/stored_queries/test_ncbi_tax.py]: What other modifiers like this are possible? Are there any that could cause problems, given we're basically exposing this part of the arango api?

File: test/stored_queries/test_ncbi_tax.py [hunk 3]
```diff
@@ -136,3 +136,52 @@ def test_siblings_nonexistent_node(self):
...
+        ).json()
+        self.assertEqual(resp['count'], 1)
```
Comment (MrCreosote) [inline test/stored_queries/test_ncbi_tax.py]: No check that the result is what you expected?

File: stored_queries/ncbi_tax/ncbi_taxon_search_sci_name.yaml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+      description: Text to search on for the scientific name
+    offset:
+      type: integer
+      default: 0
+    limit:
+      type: integer
+      default: 20
+query: |
+  FOR doc in FULLTEXT(ncbi_taxon, "scientific_name", @search_text)
+    LIMIT @offset, @limit
```
Comment (MrCreosote) [inline stored_queries/ncbi_tax/ncbi_taxon_search_sci_name.yaml]: Doesn't seem like these are tested

File: schemas/ncbi/ncbi_taxon.yaml [hunk 5]
```diff
@@ -1,5 +1,10 @@
 name: ncbi_taxon
```
Comment (MrCreosote) [inline schemas/ncbi/ncbi_taxon.yaml]: So can we ditch the canonicalization of names in the nodes then?

File: test/stored_queries/test_ncbi_tax.py [hunk 6]
```diff
@@ -136,3 +136,52 @@ def test_siblings_nonexistent_node(self):
...
+    def test_search_sciname_more_complicated(self):
+        """Test a query to search sciname."""
+        resp = requests.post(
+            _CONF['re_api_url'] + '/api/v1/query_results',
+            params={'stored_query': 'ncbi_taxon_search_sci_name'},
+            data=json.dumps({'search_text': "prefix:gamma,|prefix:alpha,|prefix:delta"})
+        ).json()
+        self.assertEqual(resp['count'], 3)
+        names = {r['scientific_name'] for r in resp['results']}
+        self.assertEqual(names, {'Gammaproteobacteria', 'Alphaproteobacteria', 'Deltaproteobacteria'})
```
Comment (MrCreosote) [inline test/stored_queries/test_ncbi_tax.py]: Would be good to have some tests on trickier names w/ multiple tokens, punctuation, capitalization etc

**Comment by jayrbolton** [comment]
Will address your comments, thanks. I am also going to change the format of the indexes config in the schema. I realized it won't work well to have the index type as the key if we want several, separate indexes of the same type. So I will make it just an array of config objects.

**Comment by jayrbolton** [comment]
Changed the format of the index config in schemas/ncbi/ncbi_taxon.yaml

File: schemas/ncbi/ncbi_taxon.yaml (hunk 5 above)
Comment (jayrbolton) [reply schemas/ncbi/ncbi_taxon.yaml]: Yeah, most likely

File: stored_queries/ncbi_tax/ncbi_taxon_search_sci_name.yaml (hunk 1 above)
Comment (jayrbolton) [reply stored_queries/ncbi_tax/ncbi_taxon_search_sci_name.yaml]: Changed this comment and added offset limit of 10k

File: test/stored_queries/test_ncbi_tax.py (hunk 2 above)
Comment (jayrbolton) [reply test/stored_queries/test_ncbi_tax.py]: The prefixes are "prefix:xyz", "complete:xyz", (the default), "|xyz" (or operator), "-xyz" (negation), "+xyz" (the default). I really don't think there would be any problems exposing this

File: test/stored_queries/test_ncbi_tax.py (hunk 3 above)
Comment (jayrbolton) [reply test/stored_queries/test_ncbi_tax.py]: Added a check on the result here

File: test/stored_queries/test_ncbi_tax.py (hunk 6 above)
Comment (jayrbolton) [reply test/stored_queries/test_ncbi_tax.py]: I think we should hold off on this until we know that the query and data is going to be used in an actual feature. For now this is just a "demo" endpoint, so having more detailed tests won't be worth it

File: stored_queries/ncbi_tax/ncbi_taxon_search_sci_name.yaml (hunk 4 above)
Comment (jayrbolton) [reply stored_queries/ncbi_tax/ncbi_taxon_search_sci_name.yaml]: Added a test for the max offset. Not going to other pagination results for now

**Pull request merged**